### PR TITLE
dev packages

### LINKED ISSUE #3995: Deploy Conan develop branch on test.pypi.org
To help us debug your issue please explain:

Hi!

My proposal here is return to use test.pypi.org (used until Conan 0.4.1) when merging a new PR into develop branch. I see no difficult installing Conan on my machine running `pip install .` but for who wants to install the latest version (non-stable) without git, it could be an good option.

For example, bincrafters-package-tools is uploaded to test.pypi.org for every new change in master branch (we follow github flow), only tags are uploaded to pypi.org. That flow helped me to find many bugs, including missing files in the package.

The version released could be incremental, using the suffix dev as well.

Conan Version: ANY
OS: ANY

- [X] I've read the [CONTRIBUTING guide](https://raw.githubusercontent.com/conan-io/conan/develop/.github/CONTRIBUTING.md).
- [X] I've specified the Conan version, operating system version and any tool that can be relevant.
- [X] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

### PR DESCRIPTION
**Pull request by lasote**
Changelog: Feature: Deploy a conan snapshot package to [test.pypi.org](https://test.pypi.org/project/conan/) for every develop commit.

Closes #3995 
I tried to use the git SHA for the commit but it is not allowed. Only number chars there, so I use a timestamp.

### CODE REVIEW

File: bump_dev_version.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import os
+import subprocess
```
Comment (memsharded) [inline bump_dev_version.py]: subprocess not needed?

File: .ci/jenkins/Jenkinsfile [hunk 2]
```diff
@@ -161,6 +161,20 @@ try{
         }*/ // EXTREMELY SLOW, INVESTIGATE
         parallel builders
     }
+    if (env.BRANCH_NAME == "develop") {
+        // Deploy snapshot to test pypi if branch develop
+        node("Linux") {
+           stage("Deploy snapshot to pypitesting"){
+              checkout scm
+              withCredentials([string(credentialsId: 'TWINE_USERNAME', variable: 'TWINE_USERNAME'),
```
Comment (uilianries) [inline .ci/jenkins/Jenkinsfile]: are you using the same credentials for test.pypi.org and pypi.org?

File: bump_dev_version.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import os
```
Comment (memsharded) [inline bump_dev_version.py]: Maybe put this script in ``.ci`` folder, to not further fill the root folder?

File: .ci/jenkins/Jenkinsfile (hunk 2 above)
Comment (lasote) [reply .ci/jenkins/Jenkinsfile]: That is about Jenkins, the user/pass are stored in a "credential" storage.

File: bump_dev_version.py (hunk 3 above)
Comment (lasote) [reply bump_dev_version.py]: ok

File: bump_dev_version.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        if search not in content:
+            raise Exception("Incorrect development version in conans/__init__.py")
+    content = content.replace(search, replace)
+    content = content.encode("utf-8")
+    with open(file_path, "wb") as handle:
+        handle.write(content)
+
+def bump_dev():
+    vfile = os.path.join(os.path.dirname(os.path.abspath(__file__)), "conans/__init__.py")
+    snapshot = "%s" % int(time.time())
```
Comment (uilianries) [inline bump_dev_version.py]: as bumping incremental version needs a new commit, date is a good option.

**Pull request merged**